src/client/datascience/jupyter/kernels/kernelExecution.ts

```diff
@@ -57,36 +55,35 @@ export class KernelExecution implements IDisposable {
     }
 
     @captureTelemetry(Telemetry.ExecuteNativeCell, undefined, true)
-    public async executeCell(cell: NotebookCell): Promise<void> {
-        if (!this.notebook) {
-            throw new Error('executeObservable cannot be called if kernel has not been started!');
-        }
+    public async executeCell(notebookPromise: Promise<INotebook>, cell: NotebookCell): Promise<void> {
         // Cannot execute empty cells.
         if (this.cellExecutions.has(cell) || cell.document.getText().trim().length === 0) {
             // clear cell output
             cell.outputs = [];
             return;
         }
+        const editor = this.vscNotebook.notebookEditors.find((item) => item.document === cell.notebook);
+        if (!editor) {
+            return;
+        }
         const cellExecution = this.executionFactory.create(cell, isPythonKernelConnection(this.metadata));
         this.cellExecutions.set(cell, cellExecution);
```

Comment:
So this line here needs to run in order to indicate the cell is executing. We used to wait for the notebook promise before we did this, but now it gets passed the notebook promise instead?

Reply:
Yes, the line is `executionFactory.create`. We need to create the `CellExecution`.
I.e. we   need to start queuing the cell execution (which happens when we create that class).